app/views/staff/rooms/create.js.erb

```diff
@@ -1,6 +1,7 @@
-$("#room-new-dialog").modal("hide");
+$("#room-new-dialog .errors").html("");
 
 <% if room.persisted? %>
```

Comment:
I noticed the call to #save below for the track modal update and left a comment there about checking the errors collection instead. #persisted? works here since it's a create operation. #valid? and #changed? would also work, though personally I would go with checking the errors collection since in my mind that connects directly with the displaying of the error messages.